src/coordsim/simulation/simulatorparams.py

```diff
@@ -106,13 +127,66 @@ def update_single_inter_arr_mean(self, new_mean):
     def update_single_predicted_inter_arr_mean(self, new_mean):
         self.predicted_inter_arr_mean = {node_id: new_mean for node_id in self.network.nodes}
 
-    # string representation for logging
-    def __str__(self):
-        params_str = "Simulator parameters: \n"
-        params_str += "inter_arr_mean: {}\n".format(self.inter_arr_mean)
-        params_str += f"deterministic_arrival: {self.deterministic_arrival}\n"
-        params_str += "flow_dr_mean: {}\n".format(self.flow_dr_mean)
-        params_str += "flow_dr_stdv: {}\n".format(self.flow_dr_stdev)
-        params_str += "flow_size_shape: {}\n".format(self.flow_size_shape)
-        params_str += f"deterministic_size: {self.deterministic_size}\n"
-        return params_str
+    def reset_flow_lists(self):
+        """Reset and re-init flow data lists and index. Called at the beginning of each new episode."""
+        # list of generated inter-arrival times, flow sizes, and data rates for the entire episode
+        # dict: ingress_id --> list of arrival times, sizes, drs
+        self.flow_arrival_list = {}
+        self.flow_size_list = {}
+        self.flow_dr_list = {}
+        self.flow_list_idx = {}
+
+        for ing_node in self.ing_nodes:
+            self.flow_arrival_list[ing_node[0]] = []
+            self.flow_size_list[ing_node[0]] = []
+            self.flow_dr_list[ing_node[0]] = []
+            # index in these lists: is initialized and reset when generating the lists
+            # dict: ingress_id --> list index
+            self.flow_list_idx[ing_node[0]] = 0
```

Comment:
It is more compact, but will run 3 more loops doing basically the same thing.

Reply:
True. I adjusted it to copy the dict where possible, eliminating 2 loops.

Either way, we only call this once per episode and we usually have very few ingress nodes to iterate over. So I don't think we would notice any performance difference.

Reply:
Sure. since we are using it once per episode it doesn't hurt. 
Btw, `copy()` is the same as iterating over the elements and copying them. It has a complexity of O(n) https://wiki.python.org/moin/TimeComplexity

Reply:
True!

Reply:
I had to revert this, I switched to the one you suggested with inline loops. The copy() method seems to be copying the pointers to the arrays inside the keys, and not creating new ones which caused crazy numbers to show up in the data rate lists.